crates/dbsp/src/storage/backend/posixio_impl.rs

```diff
@@ -72,6 +95,7 @@ impl FileReader for PosixReader {
     }
 
     fn read_block(&self, location: BlockLocation) -> Result<Arc<FBuf>, StorageError> {
+        sleep(self.ioop_delay);
```

Comment:
you probably checked that this has no overhead if duration is 0

Reply:
since you asked about it, maybe this would be better addressed having it outside our software e.g., using a script that sets up a mount point us dm-delay or similar

Reply:
> you probably checked that this has no overhead if duration is 0

I did, yes.

Reply:
> since you asked about it, maybe this would be better addressed having it outside our software e.g., using a script that sets up a mount point us dm-delay or similar

I like the idea in principle, but in practice it's going to be a lot harder to deploy, especially in environments that the developer does not entirely control.